### ISSUE TITLE
Cleanup travis-ci setup

### ISSUE THREAD

**Issue by larskuhtz** [alephcloud/hs-configuration-tools#45]
Currently the the travis-ci build matrix has 32 entries. A build can take about to 8 hours in total and we often get

```
No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
```

**Comment by larskuhtz** [comment]
PR #48 addresses this issue up to a point that travis-ci build don't timeout any more. Yet the builds take far to long. This as two reasons:

1.  The build matrix is to large (this addresses by PR #49)
2.  Building the package takes too long for ghc >= 7.8 and Cabal >= 1.20. It seems that cabal is much 
     slower when compiling the modules in configuration-tools then for the dependencies. It
     becomes slow when it compiles `Utils.Validation` and then stays slow for all remaining modules.

**Comment by larskuhtz** [comment]
Without PR #48 the build with `cabal >= 1.20` and `ghc >= 7.8` times out after hanging 10 minutes at `Configuration.Utils.Internal.ConfigFileReader`. 

An explanation could be a memory consumption issue that causes the build process to do extensive swapping. (For instance due to usage of large C-header files during the compilation.)

**Comment by larskuhtz** [comment]
Solved by PR #49
